perl/bin/cgpVaf.pl

```diff
@@ -92,54 +87,50 @@ BEGIN
         'mq'            => $vcf_obj->{'_mq'},
         'bq'            => $vcf_obj->{'_bq'},
         'exp'           => $vcf_obj->{'_exp'},
-        );
+        'emb'           => $vcf_obj->{'_emb'},
+    );
 
     foreach my $chr(keys %$chromosomes) {
         my($data_for_all_samples,$unique_locations)=$vcf_obj->getMergedLocations($chr, $vcf_file_obj);
         if(defined $options->{'b'} ){
-          my ($bed_locations)=$vcf_obj->getBedHash($chr);
+            my ($bed_locations)=$vcf_obj->getBedHash($chr);
             if( $options->{'bo'} == 1 && (defined $data_for_all_samples) ) {
                 ($data_for_all_samples,$unique_locations)=$vcf_obj->populateBedLocationsFromVCF($data_for_all_samples,$unique_locations,$bed_locations);
             }else{
-              ($data_for_all_samples,$unique_locations)=$vcf_obj->populateBedLocations($data_for_all_samples,$unique_locations,$bed_locations);
+                ($data_for_all_samples,$unique_locations)=$vcf_obj->populateBedLocations($data_for_all_samples,$unique_locations,$bed_locations);
             }
         }
-         $vcf_obj->processMergedLocations($data_for_all_samples ,$unique_locations,$variant,$chromosomes->{$chr} ,$bam_objects,$chr,$tags,$info_tag_val);
+        $vcf_obj->processMergedLocations($data_for_all_samples ,$unique_locations,$variant,$chromosomes->{$chr} ,$bam_objects,$chr,$tags,$info_tag_val);
     }# completed all chromosomes;
 
-        # run following steps only if chromosome option is empty or user has selected option to concatenate files.
+    # run following steps only if chromosome option is empty or user has selected option to concatenate files.
     if($options->{'ct'} || @{$options->{'chr'}} == 0 ) {
```

Comment:
if we remove chr check then concatenation will be done for user supplied list, provided it was not set empty on line #235

Reply:
`catFiles` doesn't take the chr into account in any way:

https://github.com/cancerit/vafCorrect/blob/5c5f3c0e6183863ee91469af070f1f10384fff37/perl/lib/Sanger/CGP/Vaf/Data/ReadVcf.pm#L1460-L1469

As stated in my comment, removing the `|| @{$options->{'chr'}} == 0` caused strange behaviour in the CI system.

My remit was to make this code function as a container and get it into the pipeline.

Reply:
ok, sounds good plan